test/model/scheduler/SchedulerConfigStorageTest.php

```diff
@@ -0,0 +1,129 @@
+        $storage = $this->getStorage();
+        $callbackMock = $this->getMockBuilder('\stdClass')
+            ->setMethods(['myCallBack'])
+            ->getMock();
+        $job = new Job('* * * * *', new DateTime('now'), $callbackMock);
+        $storage->add($job);
+        //Attempt to add the same job
+        $storage->add($job);
+    }
+
+    public function testRemove()
+    {
+        $storage = $this->getStorage();
+        $job1 = new Job('* * * * *', new DateTime('@'.time()), 'time', ['foo', 'bar']);
+        $job2 = new Job('* 2 * * *', new DateTime('@'.time()), 'time', ['foo', 'bar']);
+
+        $this->assertEquals(0, count($storage->getJobs()));
+        $this->assertTrue($storage->add($job1));
+        $this->assertTrue($storage->add($job2));
+        $this->assertEquals(2, count($storage->getJobs()));
+        $this->assertTrue($storage->remove($job2));
+        $this->assertEquals(1, count($storage->getJobs()));
+        $this->assertEquals($job1->getParams(), $storage->getJobs()[0]->getParams());
+        $this->assertEquals($job1->getStartTime(), $storage->getJobs()[0]->getStartTime());
+        $this->assertEquals($job1->getRRule(), $storage->getJobs()[0]->getRRule());
+        $this->assertEquals($job1->getCallable(), $storage->getJobs()[0]->getCallable());
+        $this->assertTrue($storage->remove($job1));
+        $this->assertEquals(0, count($storage->getJobs()));
+    }
+
+    public function testRemoveException()
+    {
+        $this->expectException(SchedulerException::class);
+        $storage = $this->getStorage();
+        $callbackMock = $this->getMockBuilder('\stdClass')
+            ->setMethods(['myCallBack'])
+            ->getMock();
+        $job = new Job('* * * * *', new DateTime('now'), $callbackMock);
+        //Attempt to remove nonexistent job
+        $storage->remove($job);
+    }
+
+    private function getLocatorMock()
+    {
+        vfsStream::setup('cache');
+        $params = [
+            'dir' => vfsStream::url('cache'),
+        ];
+        $driver = new \common_persistence_PhpFileDriver();
+        $persistence = $driver->connect('test', $params);
+        $persistenceManager = $this->getMockBuilder(PersistenceManager::class)->getMock();
+        $persistenceManager->method('getPersistenceById')->willReturn($persistence);
+        $simpleCache = new \oat\oatbox\cache\KeyValueCache(array(
+            'persistence' => 'cache'
+        ));
+        $serviceLocator = $this->getServiceLocatorMock([
+            SimpleCache::SERVICE_ID => $simpleCache,
+            PersistenceManager::SERVICE_ID => $persistenceManager
+        ]);
+        $simpleCache->setServiceLocator($serviceLocator);
+        return $serviceLocator;
+    }
+
+    /**
+     * @return SchedulerConfigStorage
+     */
+    private function getStorage()
+    {
+        $storage = new SchedulerConfigStorage();
+        $storage->setServiceLocator($this->getLocatorMock());
```

Comment:
It is not compatible with SchedulerRdsStorage constructor

Reply:
Neither with `SchedulerConfigStorage`. Could be updated easily.

Reply:
new instance of that storage created in the config file:
https://github.com/oat-sa/extension-tao-scheduler/pull/50/files#diff-745d469e0c1fb5c5c68e6f529c46e101R11
That means that to inject service manager into constructor we need to use a singletone call `ServiceManager::getServiceManager()` which is deprecated. I would like to avoid that.